Indexing/IndexManager.cpp

```diff
@@ -49,6 +65,10 @@ IndexManager::~IndexManager()
   if(_alg) {
     release(GENERATING_SUBST_TREE);
   }
+
+  if(_handler){
+    delete _handler;
+  }
 }
```

Comment:
Using a pointer for handler is useful since it allows us to run checks such as `if(_handler && _handler->...)`. Is there a particular downside to explicitly deleting the handler in the destructor?